test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightEntryAndFeedDeserializerUndeclaredTests.cs

```diff
@@ -1748,6 +1750,237 @@ public void ReadNonOpenEntryEdmUntypedPropertyInComplexTest()
         }
         #endregion
 
+        #region declared Edm.Untyped property with odata.type

```

Comment:
Can you add tests for the async API?

Reply:
This test class only contains sync test cases. We'd set up the environment/configuration for async. Can I leave it to next PR?

Reply:
Since the tests perform I/O, what if you added async tests _just_ for the new test cases you have added? The [`AsyncStream`](https://github.com/OData/odata.net/blob/master/test/FunctionalTests/Microsoft.OData.Core.Tests/AsyncStream.cs) wrapper for async tests can help ensure we don't have async regressions where async code calls synchronous code.